src/targets/gpu/target.cpp

```diff
@@ -132,14 +133,20 @@ std::vector<pass> target::get_passes(migraphx::context& gctx, const compile_opti
         dead_code_elimination{},
         normalize_ops{},
         dead_code_elimination{},
+        // 1. For int4 ops: first remove the identity op next to packed int4 data.
+        eliminate_identity{},
+        // 2. Next: (pack/unpack)_int4 handling in simplyfy_qdq.
         simplify_qdq{},
+        // 3. Next: pruning of pack_int4 related const branches.
+        propagate_constant{{}},
```

Comment:
This should not be moved up here. This should happen in optimize_module, like always. I dont think this is needed to support `--int4-wieghts`.

Reply:
(Yes, this mayn't be necessary for supporting just the `--int4-weights`.)
This was moved up due to e.g. this simple test case needs to be supported: `test/onnx/int4_const_identity_block_sz_2_qdq_test.onnx` (new file in this PR).